sim-lib/src/sim_node.rs

```diff
@@ -202,3 +202,142 @@ impl ChannelState {
         }
     }
 }
+
+/// Represents a simulated channel, and is responsible for managing addition and removal of HTLCs from the channel and
+/// sanity checks. Channel state is tracked *unidirectionally* for each participant in the channel.
+///
+/// Each node represented in the channel tracks only its outgoing HTLCs, and balance is transferred between the two
+/// nodes as they settle or fail. Given some channel: node_1 <----> node_2:
+/// * HTLC sent node_1 -> node_2: added to in-flight outgoing htlcs on node_1.
+/// * HTLC sent node_2 -> node_1: added to in-flight outgoing htlcs on node_1.
+///
+/// Rules for managing balance are as follows:
+/// * When a HTLC is in flight, the channel's local outgoing liquidity decreases (as it's locked up).
+/// * When a HTLC fails, the balance is returned to the local node (the one that it was in-flight / outgoing on).
+/// * When a HTLC succeeds, the balance is sent to the remote node (the one that did not track it as in-flight).
+///
+/// With each state transition, the simulated channel checks that the sum of its local balances and in-flight equal the
+/// total channel capacity. Failure of this sanity check represents a critical failure in the state machine.
+#[derive(Clone)]
+pub struct SimulatedChannel {
+    capacity_msat: u64,
+    short_channel_id: u64,
+    node_1: ChannelState,
+    node_2: ChannelState,
+}
+
+impl SimulatedChannel {
+    /// Creates a new channel with the capacity and policies provided. The total capacity of the channel is evenly split
+    /// between the channel participants (this is an arbitrary decision).
+    pub fn new(
+        capacity_msat: u64,
+        short_channel_id: u64,
```

Comment:
Even if we just add it as an alias without re-defining any method it would make things more readable, especially for the error enums where multiple params are passed. `ChannelNotFound(scid)` reads better than `ChannelNotFound(u64)` for instance.

Regarding proper formatting, I meant to do it like: `A:B:C` or `AxBxC`, but if you think we may not need to manually parse that often from the logs it may not be necessary.